packages/palette_generator/example/lib/main.dart

```diff
@@ -189,20 +192,21 @@ class PaletteSwatches extends StatelessWidget {
   ///
   /// The [generator] is optional. If it is null, then the display will
   /// just be an empty container.
-  const PaletteSwatches({Key key, this.generator}) : super(key: key);
+  const PaletteSwatches({Key? key, this.generator}) : super(key: key);
 
   /// The [PaletteGenerator] that contains all of the swatches that we're going
   /// to display.
-  final PaletteGenerator generator;
+  final PaletteGenerator? generator;
 
   @override
   Widget build(BuildContext context) {
     final List<Widget> swatches = <Widget>[];
-    if (generator == null || generator.colors.isEmpty) {
+    final PaletteGenerator? paletteGen = generator;
```

Comment:
Why is this variable needed: why can't `generator` be used directly?

Reply:
This is a common workaround in NNBD migrations for the fact that the analyzer assumes that a property could be dynamic and return `null` "randomly" (e.g., returns `null` every other time it's called). So without this you need `!` everywhere even though you've checked that it's non-null.

IIRC the Dart folks are planning on improving that at some point, but for now when it's more than just a single use this is generally less ugly.

Reply:
Ahh, OK. Let's hope they do improve it: that's pretty non-intuitive.